### PR TITLE
add connect accept tcp

### PR DESCRIPTION
**Pull request by achrefbensaad**

### CODE REVIEW

**Comment by achrefbensaad** [comment]
**Accept**
image
**Connect**
image

**Comment by achrefbensaad** [comment]
Discussion points:

- [x] Validate the log format
- [x] Confirm that the kprobe numbers (400 & 401) adheare to kubearmor convention).

File: KubeArmor/monitor/logUpdate.go [hunk 1]
```diff
@@ -214,6 +214,41 @@ func (mon *SystemMonitor) UpdateLogs() {
...
+				var protocol string
+				if val, ok := msg.ContextArgs[0].(string); ok {
+					protocol = val
+				}
+
+				if val, ok := msg.ContextArgs[1].(map[string]string); ok {
+					sockAddr = val
+				}
+				log.Operation = "Network"
+				log.Resource = "remote=" + sockAddr["sin_addr"] + " port=" + sockAddr["sin_port"] + " protocol=" + protocol
```
Comment (seswarrajan) [inline KubeArmor/monitor/logUpdate.go]: Hello @achrefbensaad 
Please change the resource value from remote to remoteip. 
Suggest to use the same as TCPConnect

**Comment by achrefbensaad** [comment]
**IPv6 Connect**
image
**IPv6 Accept**
image

**Review by seswarrajan — APPROVED** [review]
LGTM. @nyrahul Please provide your comments.

File: KubeArmor/BPF/system_monitor.c [hunk 2]
```diff
@@ -1296,4 +1302,140 @@ int kretprobe__listen(struct pt_regs *ctx)
...
+        sockv6.sin6_family = conn.skc_family;
+        sockv6.sin6_port = conn.skc_dport;
+        bpf_probe_read(&sockv6.sin6_addr.in6_u.u6_addr16, sizeof(sockv6.sin6_addr.in6_u.u6_addr16), conn.skc_v6_daddr.in6_u.u6_addr16);
+        args.args[1] = (unsigned long) &sockv6;
+        context.event_id = _TCP_CONNECT_v6;
+        break;
+
+    default:
+        return 0;
+        break;
```
Comment (nyrahul) [inline KubeArmor/BPF/system_monitor.c]:
```suggestion
        return 0;
```

File: KubeArmor/BPF/system_monitor.c [hunk 3]
```diff
@@ -1296,4 +1302,140 @@ int kretprobe__listen(struct pt_regs *ctx)
...
+    struct sock_common conn = READ_KERN(newsk->__sk_common);
+    struct sockaddr_in sockv4;
+    struct sockaddr_in6 sockv6;
+    sys_context_t context = {};
+    args_t args = {};
+    u64 types = ARG_TYPE0(STR_T)|ARG_TYPE1(SOCKADDR_T);
+    init_context(&context);
+    context.argnum = get_arg_num(types);
+
+    switch (conn.skc_family)
```
Comment (nyrahul) [inline KubeArmor/BPF/system_monitor.c]: This piece of code seems to be common between connect/accept. Can we refactor it?

File: KubeArmor/monitor/logUpdate.go [hunk 4]
```diff
@@ -214,6 +214,41 @@ func (mon *SystemMonitor) UpdateLogs() {
...
+
+				if val, ok := msg.ContextArgs[1].(map[string]string); ok {
+					sockAddr = val
+				}
+
+				log.Operation = "Network"
+				log.Resource = "remoteip=" + sockAddr["sin_addr"] + " port=" + sockAddr["sin_port"] + " protocol=" + protocol
+				log.Data = "kprobe=tcp_connect domain=" + sockAddr["sa_family"]
+
+			case TCPAccept, TCPAcceptv6:
```
Comment (nyrahul) [inline KubeArmor/monitor/logUpdate.go]: This seems almost duplicate of TCPConnect/6 above. Can we refactor the code?

File: KubeArmor/monitor/syscallParser.go [hunk 5]
```diff
@@ -188,6 +188,48 @@ func readSockaddrFromBuff(buff io.Reader) (map[string]string, error) {
...
+
+			           struct in6_addr {
+							union {
+								__u8		u6_addr8[16];
+						#if __UAPI_DEF_IN6_ADDR_ALT
+								__be16		u6_addr16[8];
+								__be32		u6_addr32[4];
+						#endif
+							} in6_u;
+		*/
```
Comment (nyrahul) [inline KubeArmor/monitor/syscallParser.go]:
```suggestion
		// https://man7.org/linux/man-pages/man7/ipv6.7.html
```

File: KubeArmor/BPF/system_monitor.c [hunk 6]
```diff
@@ -1296,4 +1302,140 @@ int kretprobe__listen(struct pt_regs *ctx)
...
+    else
+        // pre-4.10 with big endian
+        protocol = READ_KERN(*(u8 *)((u64)&newsk->sk_wmem_queued - 1));
+#else
+# error "Fix your compiler's __BYTE_ORDER__?!"
+#endif   
+  
+    if (protocol != IPPROTO_TCP)
+        return 0;
+
```
Comment (nyrahul) [inline KubeArmor/BPF/system_monitor.c]:
```suggestion
```

File: KubeArmor/BPF/system_monitor.c [hunk 7]
```diff
@@ -1296,4 +1302,140 @@ int kretprobe__listen(struct pt_regs *ctx)
...
+    sys_context_t context = {};
+    args_t args = {};
+    u64 types = ARG_TYPE0(STR_T)|ARG_TYPE1(SOCKADDR_T);
+    init_context(&context);
+    context.argnum = get_arg_num(types);
+
+    switch (conn.skc_family)
+    {
+    case AF_INET:
+
```
Comment (nyrahul) [inline KubeArmor/BPF/system_monitor.c]:
```suggestion
```

File: KubeArmor/BPF/system_monitor.c [hunk 8]
```diff
@@ -1296,4 +1302,140 @@ int kretprobe__listen(struct pt_regs *ctx)
...
+        sockv6.sin6_family = conn.skc_family;
+        sockv6.sin6_port = (conn.skc_num>>8) | (conn.skc_num<<8);
+        bpf_probe_read(&sockv6.sin6_addr.in6_u.u6_addr16, sizeof(sockv6.sin6_addr.in6_u.u6_addr16), conn.skc_v6_daddr.in6_u.u6_addr16);
+        args.args[1] = (unsigned long) &sockv6;
+        context.event_id = _TCP_ACCEPT_v6;
+        break;
+
+    default:
+        return 0;
+        break;
```
Comment (nyrahul) [inline KubeArmor/BPF/system_monitor.c]:
```suggestion
        return 0;
```

**Review by nyrahul — CHANGES_REQUESTED** [review]

File: KubeArmor/BPF/system_monitor.c (hunk 3 above)
Comment (achrefbensaad) [reply KubeArmor/BPF/system_monitor.c]: Done

File: KubeArmor/monitor/logUpdate.go (hunk 4 above)
Comment (achrefbensaad) [reply KubeArmor/monitor/logUpdate.go]: Done

**Review by nyrahul — APPROVED** [review]
LGTM

**Pull request merged**